src/modinv64_impl.h

```diff
@@ -0,0 +1,360 @@
+/**********************************************************************
+ * Copyright (c) 2020 Peter Dettman                                   *
+ * Distributed under the MIT software license, see the accompanying   *
+ * file COPYING or http://www.opensource.org/licenses/mit-license.php.*
+ **********************************************************************/
+
+
+#ifndef SECP256K1_MODINV64_IMPL_H
+#define SECP256K1_MODINV64_IMPL_H
+
+#include "modinv64.h"
+
+#include "util.h"
+
+static void secp256k1_modinv64_signed62_verify(const secp256k1_modinv64_signed62* a) {
+    /* a must be in the range [-2^256, 2^256). */
+    VERIFY_CHECK(a->v[0] >> 62 == 0);
+    VERIFY_CHECK(a->v[1] >> 62 == 0);
+    VERIFY_CHECK(a->v[2] >> 62 == 0);
+    VERIFY_CHECK(a->v[3] >> 62 == 0);
```

Comment:
This isn't true of the field prime constants using this type. I suggest that the restriction is one that applies to d, e, f, g with respect to the _update methods, but not for all instances of the type.

Reply:
What is special about the field prime constants? The tests seem to pass.

Reply:
They contain negative limbs (to maximise the number of zeros in their representation).